pkg/controller/util/deletionannotation.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// DeleteOptionAnnotation contains options for delete
+	// while deleting resources for member clusters.
+	DeleteOptionAnnotation = "kubefed.io/deleteoption"
+)
+
+// GetDeleteOptions return delete options from the annotation
+func GetDeleteOptions(obj *unstructured.Unstructured) ([]client.DeleteOption, error) {
+	options := make([]client.DeleteOption, 0)
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		return options, nil
+	}
+
+	if optStr, ok := annotations[DeleteOptionAnnotation]; ok {
+		opt := &metav1.DeleteOptions{}
+		if err := json.Unmarshal([]byte(optStr), opt); err != nil {
+			return nil, fmt.Errorf("could not deserialize delete options from annotation value '%s': %w", optStr, err)
+		}
+		clientOpt := &client.DeleteOptions{}
+		clientOpt.GracePeriodSeconds = opt.GracePeriodSeconds
+		clientOpt.PropagationPolicy = opt.PropagationPolicy
+		clientOpt.Preconditions = opt.Preconditions
```

Comment:
To be strict here, you are missing `DryRun` and Raw fields here.

Reply:
@hectorj2f , 
1. I don't think `DryRun`  should be applied here. Deletion will be called infinitely when applied.
2. `Raw` will be overridden in controller-runtime. This field should be ignored. 
The client.DeleteOptions if a wrap of the metav1.DeleteOptions. So I only picked up those 3 fields that currently available.

Reply:
This is gonna be problematic in the future with this manual variable assignations. Attributes could be missed in the future.

Reply:
Let's defer this action for now. Not a blocker.